kmir/src/kmir/kdist/mir-semantics/rt/data.md

```diff
@@ -1315,6 +1398,57 @@ The `unOpNot` operation works on boolean and integral values, with the usual sem
 `binOpShr`
 `binOpShrUnchecked`
 
+```k
+  syntax Bool ::= isBitwise ( BinOp ) [function, total]
+  // --------------------------------------------------
+  rule isBitwise(binOpBitXor) => true
+  rule isBitwise(binOpBitAnd) => true
+  rule isBitwise(binOpBitOr)  => true
+  rule isBitwise(_)           => false [owise]
+
+  syntax Bool ::= isShift ( BinOp ) [function, total]
+  // ------------------------------------------------
+  rule isShift(binOpShl)          => true
+  rule isShift(binOpShlUnchecked) => true
+  rule isShift(binOpShr)          => true
+  rule isShift(binOpShrUnchecked) => true
+  rule isShift(_)                 => false [owise]
+
+  rule onInt(binOpBitXor, X, Y)       => X xorInt Y
+  rule onInt(binOpBitAnd, X, Y)       => X &Int Y
+  rule onInt(binOpBitOr, X, Y)        => X |Int Y
+
+  // Only permitting non-shift bitwise operations on same width integers, overflow check not required
+  rule #compute(
+          BOP,
+          typedValue(Integer(ARG1, WIDTH, _), _, _),
+          typedValue(Integer(ARG2, WIDTH, _), _, _),
+          false) // unchecked
+    =>
+       typedValue(
+          Integer(onInt(BOP, ARG1, ARG2), WIDTH, false),
+          TyUnknown,
+          mutabilityNot
+        )
```

Comment:
Two things to correct:
* Your result is always unsigned but we have to keep the signedness flag from the arguments. 
```suggestion
  rule #compute(
          BOP,
          typedValue(Integer(ARG1, WIDTH, SIGNED), TY, _),
          typedValue(Integer(ARG2, WIDTH, SIGNED), TY, _),
          false) // unchecked
    =>
       typedValue(
          Integer(onInt(BOP, ARG1, ARG2), WIDTH, SIGNED),
          TY,
          mutabilityNot
        )
```
* We have to ensure the `Ty` are the same for both arguments (I added the match on `TY` above). There are rules that checks this, and the suitability of the operands, for arithmetic operations. Maybe we should add similar rules for bit-oriented ones, to signal `TypeMismatch` or `OperandMismatch` errors